app/usecase/event/interface.go

```diff
@@ -0,0 +1,34 @@
+package event
+
+type Dispatcher interface {
+	Emitter
+	Subscriber
+}
+
+// Emitter provides the ability to propagate the given event to the list of corresponding listeners
+type Emitter interface {
+	// Dispatch dispatches an event to all registered listeners
+	Dispatch(event Event) error
+}
+
+// Listener is an entity which handles received events
+type Listener interface {
+	// Handle processes the received event
+	Handle(event Event)
+	// GetSubscribedEvents returns the list of event names this listener wants to listen to
+	GetSubscribedEvents() []string
```

Comment:
I wonder whether we should subscribe multiple events at the same time?

Reply:
This doesn't seem be necessary.

Reply:
You are right probably. I will change it:)